components/SingleEntry/Entry.js

```diff
@@ -90,22 +104,37 @@ export default function Entry({ id }) {
           </Container>
         )}
       </AnimatePresence>
-      <ToolsContainer>
-        {/* <DeleteButton as="a" onClick={() => router.back()}>
-          {showSentence ? (
-            <FiArrowLeft />
-          ) : (
-            <FiArrowLeft style={{ color: "grey", opacity: "0.5" }} />
-          )}
-        </DeleteButton> */}
-        <DeleteButton as="a" onClick={() => handleShowSentence()}>
-          {showSentence ? (
-            <FaRegEyeSlash />
+      {session && (
+        <ToolsContainer>
+          {deletingId === entry._id ? (
+            <DeleteButton
+              style={{ color: "red" }}
+              as="a"
+              onClick={(event) => handleDeleteEntry(event, entry._id)}
+            >
+              <FiTrash2 />
+            </DeleteButton>
           ) : (
-            <FaRegEye style={{ color: "grey", opacity: "0.5" }} />
+            <DeleteButton
+              as="a"
+              onClick={(event) => handleDeleteDialog(event, entry._id)}
+            >
+              {showSentence ? (
+                <FiTrash2 />
+              ) : (
+                <FiTrash2 style={{ color: "grey", opacity: "0.5" }} />
```

Comment:
please don't use inline styling. This can be solved with a prop (e.g. $isDimmed or $disabled)